src/libasr/codegen/asr_to_llvm.cpp

```diff
@@ -4769,6 +4769,73 @@ class ASRToLLVMVisitor : public ASR::BaseVisitor<ASRToLLVMVisitor>
                 tmp = builder->CreateSelect(cmp, zero_str, one_str);
                 break;
             }
+            case (ASR::cast_kindType::CharacterToList) : {
+                this->visit_expr_wrapper(x.m_arg, true);
```

Comment:
I think this line is not required. At the start of the switch case (`Line: 4520`), it's already been called, so you can simply use `tmp` to get the value.